cinema/src/main/java/com/greg/cinema/mapper/ReservationMapper.java

```diff
@@ -0,0 +1,38 @@
+package com.greg.cinema.mapper;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import com.greg.cinema.constant.TimeConstants;
+import com.greg.cinema.dto.ReservationRequest;
+import com.greg.cinema.dto.ReservationResponse;
+import com.greg.cinema.entity.ReservationEntity;
+import com.greg.cinema.entity.ScreeningEntity;
+import com.greg.cinema.entity.SeatReservedEntity;
+
+@Component
+public class ReservationMapper {
+	
+	public ReservationEntity toEntity(ReservationRequest dto, ScreeningEntity screening) {
+		ReservationEntity entity = new ReservationEntity();
+		
+		entity.setFirstName(dto.getFirstName());
+		entity.setLastName(dto.getLastName());
+		entity.setScreening(screening);
+		entity.setCreateTime(LocalDateTime.now());
+		entity.setExpireTime(entity.getCreateTime().plusMinutes(TimeConstants.RESERVATION_EXPIRY_TIME));
+		
+		return entity;
+	}
+	
+	public ReservationResponse toDto(ReservationEntity reservation, List<SeatReservedEntity> seats) {
+		double cost = seats.stream().mapToDouble(s -> s.getTicketType().getPrice()).sum();
```

Comment:
double to też nie polecany typ przy $$ - wiesz dlaczego?

Reply:
Jak wyżej, użyłem float/double dla prostoty